GoogleUtilities/Example/Tests/Logger/GULASLLoggerTest.m

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <GoogleUtilities/GULASLLogger.h>
+#import <GoogleUtilities/GULLogger.h>
+
+#import <asl.h>
+
+#import <XCTest/XCTest.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// TODO(bstpierre): Use a C function redirect to mock asl_* methods like GULOSLoggerTest.
+
+static NSString *const kService = @"my service";
+static NSString *const kCode = @"I-COR000001";
+
+// Redefine class property as readwrite for testing.
+@interface GULLogger (ForTesting)
+@property(nonatomic, class, readwrite) id<GULLoggerSystem> logger;
+@end
+
+// Surface aslclient and dispatchQueues for tests.
+@interface GULASLLogger (ForTesting)
+@property(nonatomic) aslclient aslClient;
+@property(nonatomic) dispatch_queue_t dispatchQueue;
+@end
+
+#pragma mark -
+
+@interface GULASLLoggerTest : XCTestCase
+@property(nonatomic) GULASLLogger *logger;
+@end
+
+@implementation GULASLLoggerTest
+
+#pragma mark Helper Methods
+
+// TODO(bstpierre): Replace this with a XCTestExpectation like GULOSLoggerTest.
+- (BOOL)messageWasLogged:(NSString *)message {
```

Comment:
I don't see this method is used anymore -  should we be validating the ASLLogger here or is there enough coverage elsewhere?

Reply:
Added #2424 to track adding this stubbing and testing.